### PR TITLE
fix/rendezvous_connect: fix weird packet loss issue

### PR DESCRIPTION
**Pull request by canndrew**

### CODE REVIEW

**Comment by povilasb** [comment]
`rustfmt` fails :/

File: src/udp/socket.rs [hunk 1]
```diff
@@ -141,21 +143,38 @@ pub trait UdpSocketExt {
...
-        socket.reuse_address(true)?;
+        let socket = bind_reusable(addr)?;
+        let socket = UdpSocket::from_socket(socket, handle)?;
 
-        #[cfg(target_family = "unix")]
-        {
-            use net2::unix::UnixUdpBuilderExt;
-            socket.reuse_port(true)?;
-        }
+        Ok(socket)
```
Comment (povilasb) [inline src/udp/socket.rs]: isn't his redundant? `UdpSocket::from_socket()` already returns `io::Result<UdpSocket>`: https://docs.rs/tokio-core/0.1.11/tokio_core/net/struct.UdpSocket.html#method.from_socket

Comment (canndrew) [reply src/udp/socket.rs]: Yeah, but I sometimes find it clearer to write like that. I've removed it though.

Comment (povilasb) [reply src/udp/socket.rs]: Hmm, personally for me, function signature is clear enough: `fn bind_reusable(addr: &SocketAddr, handle: &Handle) -> io::Result<UdpSocket> {`

**Review by povilasb — APPROVED** [review]

**Review by ustulation — APPROVED** [review]

**Pull request merged**